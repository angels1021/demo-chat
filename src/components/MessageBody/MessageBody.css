.message-body {
  position: relative;
  max-width: 80%;
  min-width: 8rem;
  background-color: ivory;
}

.message-body.bs-popover-right .arrow::after {
  border-right-color: ivory;
}

.message-body.my-message {
  background-color: #c5fbbf;
}

.message-body.my-message.bs-popover-left .arrow::after {
  border-left-color: #c5fbbf;
}
