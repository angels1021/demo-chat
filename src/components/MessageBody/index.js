/**
 * MessageText component
 *
 * @module components/MessageText
 * @component MessageText
 * @typicalname MessageText
 *
 * @requires [React]{@link https://facebook.github.io/react/docs/hello-world.html}
 * @requires [prop-types]{@link https://www.npmjs.com/package/prop-types}
 * */
import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import './MessageBody.css';

export const positions = {
  right: 'bs-popover-right',
  left: 'bs-popover-left'
};

const MessageBody = ({ children, username, isSelf }) => (
  <div className={cn({
    'message-body': true,
    'my-message': isSelf,
    popover: true,
    [positions.right]: !isSelf,
    [positions.left]: isSelf
  })}>
    <div className="px-2 py-1" >
      <div className="arrow" />
      <h6 className="text-capitalize text-info" >{(isSelf ? 'Me' : username)}</h6>
      <div>{children}</div>
    </div>
  </div>
);

MessageBody.propTypes = {
  children: PropTypes.node.isRequired,
  username: PropTypes.string.isRequired,
  isSelf: PropTypes.bool.isRequired
};

export default MessageBody;