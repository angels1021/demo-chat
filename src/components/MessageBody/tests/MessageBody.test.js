import React from 'react';
import { shallow } from 'enzyme';
import MessageBody, { positions } from '../';

const childrenNode = <h1>hello</h1>;
const render = (isSelf = false) => shallow(
  <MessageBody username="tom" isSelf={isSelf} >
    {childrenNode}
  </MessageBody>
);

describe('MessageBody', () => {
  describe('message by other user', () => {
    const renderedComponent = render();

    it('should return the match snapshot', () => {
      expect(renderedComponent).toMatchSnapshot();
    });

    it('should contain passed in children', () => {
      // assert
      expect(renderedComponent.contains(childrenNode)).toBe(true);
    });

    it('should add the correct classes', () => {
      expect(renderedComponent.hasClass(positions.right)).toBe(true);
      expect(renderedComponent.hasClass(positions.left)).toBe(false);
      expect(renderedComponent.hasClass('my-message')).toBe(false);
    });

    it('should print the correct username', () => {
      const username = renderedComponent.find('.text-info').first().text();
      expect(username).toBe('tom');
      expect(username).not.toBe('Me');
    });

  });

  describe('message by the current user', () => {
    const renderedComponent = render(true);

    it('should return the match snapshot', () => {
      expect(renderedComponent).toMatchSnapshot();
    });

    it('should add the correct classes', () => {
      expect(renderedComponent.hasClass(positions.right)).toBe(false);
      expect(renderedComponent.hasClass(positions.left)).toBe(true);
      expect(renderedComponent.hasClass('my-message')).toBe(true);
    });

    it('should print the correct username', () => {
      const username = renderedComponent.find('.text-info').first().text();
      expect(username).not.toBe('tom');
      expect(username).toBe('Me');
    });

  });
});
