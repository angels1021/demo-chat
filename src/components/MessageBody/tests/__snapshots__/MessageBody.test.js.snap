// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageBody message by other user should return the match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MessageBody
    isSelf={false}
    username="tom"
>
    <h1>
        hello
    </h1>
</MessageBody>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="px-2 py-1"
>
        <div
                className="arrow"
        />
        <h6
                className="text-capitalize text-info"
        >
                tom
        </h6>
        <div>
                <h1>
                        hello
                </h1>
        </div>
</div>,
      "className": "message-body popover bs-popover-right",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="arrow"
/>,
          <h6
            className="text-capitalize text-info"
>
            tom
</h6>,
          <div>
            <h1>
                        hello
            </h1>
</div>,
        ],
        "className": "px-2 py-1",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "arrow",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "tom",
            "className": "text-capitalize text-info",
          },
          "ref": null,
          "rendered": "tom",
          "type": "h6",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              hello
</h1>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "hello",
            },
            "ref": null,
            "rendered": "hello",
            "type": "h1",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="px-2 py-1"
>
          <div
                    className="arrow"
          />
          <h6
                    className="text-capitalize text-info"
          >
                    tom
          </h6>
          <div>
                    <h1>
                              hello
                    </h1>
          </div>
</div>,
        "className": "message-body popover bs-popover-right",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="arrow"
/>,
            <h6
              className="text-capitalize text-info"
>
              tom
</h6>,
            <div>
              <h1>
                            hello
              </h1>
</div>,
          ],
          "className": "px-2 py-1",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "arrow",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "tom",
              "className": "text-capitalize text-info",
            },
            "ref": null,
            "rendered": "tom",
            "type": "h6",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h1>
                hello
</h1>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "hello",
              },
              "ref": null,
              "rendered": "hello",
              "type": "h1",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`MessageBody message by the current user should return the match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MessageBody
    isSelf={true}
    username="tom"
>
    <h1>
        hello
    </h1>
</MessageBody>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="px-2 py-1"
>
        <div
                className="arrow"
        />
        <h6
                className="text-capitalize text-info"
        >
                Me
        </h6>
        <div>
                <h1>
                        hello
                </h1>
        </div>
</div>,
      "className": "message-body my-message popover bs-popover-left",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="arrow"
/>,
          <h6
            className="text-capitalize text-info"
>
            Me
</h6>,
          <div>
            <h1>
                        hello
            </h1>
</div>,
        ],
        "className": "px-2 py-1",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "arrow",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Me",
            "className": "text-capitalize text-info",
          },
          "ref": null,
          "rendered": "Me",
          "type": "h6",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              hello
</h1>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "hello",
            },
            "ref": null,
            "rendered": "hello",
            "type": "h1",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="px-2 py-1"
>
          <div
                    className="arrow"
          />
          <h6
                    className="text-capitalize text-info"
          >
                    Me
          </h6>
          <div>
                    <h1>
                              hello
                    </h1>
          </div>
</div>,
        "className": "message-body my-message popover bs-popover-left",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="arrow"
/>,
            <h6
              className="text-capitalize text-info"
>
              Me
</h6>,
            <div>
              <h1>
                            hello
              </h1>
</div>,
          ],
          "className": "px-2 py-1",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "arrow",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Me",
              "className": "text-capitalize text-info",
            },
            "ref": null,
            "rendered": "Me",
            "type": "h6",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h1>
                hello
</h1>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "hello",
              },
              "ref": null,
              "rendered": "hello",
              "type": "h1",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
